sdks/java/harness/src/main/java/org/apache/beam/fn/harness/status/MemoryMonitor.java

```diff
@@ -194,21 +196,35 @@ public long totalGCTimeMilliseconds() {
 
   private final File localDumpFolder;
 
+  private final boolean gzipCompress;
+
   public static MemoryMonitor fromOptions(PipelineOptions options) {
     SdkHarnessOptions sdkHarnessOptions = options.as(SdkHarnessOptions.class);
-    String uploadFilePath = options.getTempLocation();
+    @Nullable String uploadFilePath = sdkHarnessOptions.getRemoteHeapDumpLocation();
+    if (uploadFilePath == null) {
+      uploadFilePath = options.getTempLocation();
```

Comment:
Would be useful to have defensive tests of this code path. (aka a couple tests of `fromOptions` with different options set)